### DISCUSSION TITLE
ZuluSCSI drive name

### DISCUSSION THREAD

**Discussion by phe78** [ZuluSCSI/ZuluSCSI-firmware#285] [General]
Hi there, first of all, congrats to the team for a great device : I installed it in a E-MU E6400 sampler and it works wonderfully, both with images and in raw mode. 
One question or suggestion of improvement : all images are displayed on the E-MU as a drive named "Dx ZuluscsiHARDDRIV" where x is the SCSI id. This makes it difficult to check which image file relates to which drive, would that be possible that the drive label is set to the image file (truncated if too long) instead of the fixed text "ZuluscsiHARDDRIV" ? 
Maybe I am missing something and this can be done by means of the init file ?
thanks in advance for your feedback,
Philippe 
![IMG_0476](https://github.com/ZuluSCSI/ZuluSCSI-firmware/assets/15370398/75c6e5ae-9462-47b7-8391-e5e09e2cf6eb)

**Comment by aperezbios** [comment]
@phe78 it's possible to change the product and vendor strings returned by ZuluSCSI on a per-SCSI-ID basis, using a simple zuluscsi.ini file, like so. Does this answer your question sufficiently?
```
[SCSI1]
Vendor = "ZuluSCSI"
Product = "HARDDRIVE"

[SCSI2]
Vendor = "ZuluSCSI"
Product = "HARDDRIVE"
```

**Reply by phe78** [reply]
Thanks Alex for your prompt reply ! Yes, I checked that approach, and it partially addresses the point as it is “static”. Ideally, could the firmware set the “product” variable using the name of the related image, ie at mount time rather at at init time ? That would be super useful when users are changing the contents of their image directory, and even more when ejecting and rotating the image files in a virtual CD-ROM drive : the user may read the actual name of the online image file. SCSI is probably not able to detect drive name changes in real time so that would probably need to restart the SCSI mounting process. In the case of the E-MU sampler I use, it is a matter if simply hitting “mount”, which is what I do anyway when I reinsert an updated SD card. What do you think ? Philippe

**Reply by phe78** [reply]
Hi Alex, I made a quick code change to implement this : have the vendor / product name defined by the image file name. I added a quirk (9) to reflect this behaviour a bit specific to EMU samplers. As the drive and product info displayed is limited to 8 chars each - which is a shame as lots of display space is unused - both are filled by consecutive segments of the file name (where the path, HDx_ header and extension have been removed). Only a few lines of code in ZuluSCSI_disk.cpp let me know if that is of any interest. Kind regards, Philippe
![IMG_0486](https://github.com/ZuluSCSI/ZuluSCSI-firmware/assets/15370398/f818314d-673c-48ec-b777-dee62a92221a)

**Comment by aperezbios** [comment]
Sure, please submit it as a PR and we will review it. We may refactor it before merging. I agree this is a useful feature.

**Comment by phe78** [comment]
thanks, just done. Hopefully I did not screw up, this is the 1st time I do this....
